library/pk_wrap.c

```diff
@@ -1650,6 +1650,7 @@ static int ecdsa_opaque_can_do(mbedtls_pk_type_t type)
  * using the same function. */
 #define ecdsa_opaque_check_pair_wrap    eckey_check_pair_wrap
 #else /* MBEDTLS_PK_USE_PSA_EC_DATA */
+#if defined(MBEDTLS_ECP_LIGHT)
```

Comment:
I'm not sure I understand. We in the `!MBEDTLS_PK_USE_PSA_EC_DATA` case, which means one of the following is true:
- `MBEDTLS_USE_PSA_CRYPTO` is not defined - but if that were the case we wouldn't have opaque keys at all;
- `MBEDTLS_ECP_C` is defined - if that's the case, then so is `ECP_LIGHT`, and we don't need the extra `#if` here;
- `MBEDTLS_ECP_LIGHT` is not defined - if that's the case, then this `#if` will never be true.

Perhaps the fact that it looks like we need this guard here means we've missed another one somewhere?

Or did I get myself lost in the logic? I'm looking at the following definition of `MBEDTLS_PK_USE_PSA_EC_DATA`:
```c
#if defined(MBEDTLS_USE_PSA_CRYPTO) && !defined(MBEDTLS_ECP_C) && \
    defined(MBEDTLS_ECP_LIGHT)
#define MBEDTLS_PK_USE_PSA_EC_DATA
#endif /* MBEDTLS_USE_PSA_CRYPTO && !MBEDTLS_ECP_C */
```

Reply:
Oh wait, I just realized that this definition of `MBEDTLS_PK_USE_PSA_EC_DATA` is outdated compared to the one in `development`. This was changed in #7453. That PR also removed this auto-enabler of `ECP_LIGHT`:
```
(defined(MBEDTLS_PK_C) && defined(MBEDTLS_USE_PSA_CRYPTO) && defined(PSA_WANT_ALG_ECDSA))
```
I suspect that without this auto-enabler also some other function that I managed in this PR for ECDSA will not work properly. I think that the best action will be to rebase on `development` and the test again.

Reply:
Ah, I also noticed the definition of `MBEDTLS_PK_USE_PSA_EC_DATA` seemed outdated, but then was distracted by something else and forgot to follow that thread. Thanks for rebasing to resolve that!